### ISSUE TITLE
mainline-ppa no longer supplies CHANGES, use upstream ChangeLog-#.#.# instead.

### ISSUE THREAD

**Issue by bkw777** [bkw777/mainline#48]
kernel.ubuntu.com mainline-ppa no longer supplies these changelog summaries:
https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.6.17/CHANGES

Same info is available from upstream:
https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.6.17

Change the Changes button to load that instead.

**Comment by bkw777** [comment]
Moved from #43

**Comment by StatusCode404** [comment]
I'm fine even if it was the verbose one to be honest.  As long as there's something.  The old "headlines" only was ok.  

But being analytical like I am, I sometimes wonder about the finer detail.

**Comment by tovam** [comment]
Same here, the upstream changelog is fine
There isn't any for release candidates though whereas the Ubuntu repo provided them

**Comment by bkw777** [comment]
The -rc info maybe has to come right from git, ugh...
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/log/?h=v5.8-rc4&showmsg=1

**Comment by tovam** [comment]
Yeah and cgit can't even export in plain text nor can it show logs strictly between two tags...

**Comment by noah1510** [comment]
I noticed the commit messages are all the line that contain a `<tr>` and `</tr>` tag in a single row.
It might be possible to just get those lines resulting in [this](https://gist.github.com/noah1510/ee3c20c87fb3834e032b4232131978dc) structured table which should not be too hard to parse into plain text

**Comment by bkw777** [comment]
How does one retreive that data? Is there any reasonably scalable way for each individual end user's copy of the program to fetch it efficiently by an http request? Issuing a git command is probably out of the question. Even hitting that web-accessible git commit log for the -rc kernels seems like an abuse of that web site.

kernel.ubuntu.com is set up intentionally for end-users to hit it as a distribution source for those packages, and it's a very cheap static get request to serve up those static changes files. Hitting a git repo or a web front end to a git repo to pull the commit log, from a site only intended for some developers and distro maintainers, is something else. I note the "cdn" in the hostname, so maybe there is no problem hitting it, but I still feel the app should be well-behaved and not abuse things just because they are publicly accessable and it's possible.

**Comment by StatusCode404** [comment]
Perhaps a launchpad bug may be required somewhere to get the ubuntu 
kernel guys to wake up.

**Comment by bkw777** [comment]
kernel-team says it wasn't intentional and the files are now generating again for new kernels, so we don't have to do anything.
ex: https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.8-rc5/CHANGES
